src/keys/bip39/mod.rs

```diff
@@ -0,0 +1,750 @@
+// Bitcoin Dev Kit
+// Written in 2020 by Alekos Filini <alekos.filini@gmail.com>
+//
+// Copyright (c) 2020-2021 Bitcoin Dev Kit Developers
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+//! BIP-0039
+
+// TODO: maybe write our own implementation of bip39? Seems stupid to have an extra dependency for
+// something that should be fairly simple to re-implement.
+
+use std::fmt;
+
+use bitcoin::{util::bip32, hashes::{Hash, sha256}};
+use bitcoin::Network;
+use hmac::Hmac;
+use sha2::Sha512;
+use unicode_normalization::UnicodeNormalization;
+
+use miniscript::ScriptContext;
+
+use super::{
+    any_network, DerivableKey, DescriptorKey, ExtendedKey, GeneratableKey, GeneratedKey, KeyError,
+};
+
+mod wordlists;
+pub use wordlists::Language;
+
+type Seed = [u8; 64];
+
+/// To generate a mnemonic from an entropy, the entropy must be divisble by 32 bits
+const ENTROPY_MULTIPLE: usize = 32;
+/// Minimum entropy size in bits to generate a mnemonic
+const MIN_ENTROPY_BITS: usize = 128;
+/// Maximum entropy size in bits to generate a mnemonic
+const MAX_ENTROPY_BITS: usize = 256;
+/// Mnemonic words are indices in a wordlist which are formed from 11 bits each
+const BITS_PER_WORD: usize = 11;
+/// Minimum word count a mnemonic can have
+const MIN_WORDS: usize = 12;
+/// Maximum word count a mnemonic can have
+const MAX_WORDS: usize = 24;
+
+/// A BIP39 error
+#[derive(Debug)]
+pub enum Error {
```

Comment:
While running the test on your patch, I had the following error:
![test-fail-bdk](https://user-images.githubusercontent.com/11140070/168699868-0608e31f-266a-4223-83e3-89dfb048b333.png)

This can be solved by implementing the `std::error::Error` trait on the error like so:
```suggestion
impl std::error::Error for Error{}
```

Reply:
Fixed.